Enable automatically spatial and time coordinates detected for xarray dataset

dobedobedo opened the pull request:
Use odc.geo for spatial dimension detection
doy and time is retrieved using time dimension

dobedobedo committed:
Enable automatically spatial and time coordinates detected for xarray dataset

Use odc.geo for spatial dimension detection
doy and time is retrieved using time dimension

dobedobedo commented:
Need to add odc-geo as an extra dependency

dobedobedo committed:
Update requirements.txt

Add odc-geo as extra dependency

dobedobedo committed:
Create test xarray Dataset with meaningful dimensions

dobedobedo committed:
Correct the coordinates inputs

dobedobedo committed:
Update test_geeet.py

Use the specified doy, time lat, and lon.

dobedobedo committed:
Update test_geeet.py

Change how the values are extracted when evaluation.

dobedobedo commented:
test_geeet was modified to set up the xarray dataset with meaningful coordinates.

lopezvoliver commented on geeet/tseb.py:
```diff
@@ -22,6 +22,7 @@
 
 import numpy as np
 from geeet.common import is_img, is_xr
+from odc.geo import xr # Use for automatically identify spatial dimensions
```
For now, the goal is to keep `geeet` with a very minimal requirement -- only numpy, unless absolutely necessary.

Note that `ee` is imported in a try/except statement -- (TODO: add as an optional requirements in `setup.py`) 

Similarly, `xarray` is not explicitly imported here. The plan is to also have `xarray` (and now `odc_geo`) as optional requirements, e.g. `pip install geeet[xr]`. If you want to have a go at this in `setup.py`, that would be great. Otherwise, for now you can just move it from `requirements.txt` to `requirements_dev.txt`. 

However, here we will need to move it to a try/except statement.

dobedobedo replied:
I removed odc.geo and replicate the similar function in geeet.common. Now no extra module is needed.

lopezvoliver commented on requirements.txt:
```diff
@@ -1 +1,2 @@
-numpy
\ No newline at end of file
+numpy
+odc-geo
```
Please move this to `requirements_dev.txt` for now, or better yet, update `setup.py` to include a "xarray" optional dependencies section.

dobedobedo replied:
It doesn't need odc.geo anymore so the requirements stay the same as it is.

lopezvoliver reviewed (changes requested):
Thanks for this, please check the comments about adding `odc-geo` as an optional dependency. If you can, please update `setup.py` to achieve this. I'm also planning on modernizing this to `pyproject.toml` instead of setup.py.. eventually 👀.

dobedobedo committed:
Update requirements.txt

Remove the odc.geo dependency

dobedobedo committed:
Add find_spatial_dims function

Try to replicate how odc.geo find spatial dimensions without extra module

dobedobedo committed:
Remove the need for odc.geo module

Used the new find_spatial_dims function from geeet.common to identify spatial dimensions

dobedobedo committed:
import Tuple from typing

dobedobedo commented:
Hi, I fixed the dependency problem and replicate the function in geeet.common.

lopezvoliver committed:
Allow kwarg override longitude and latitude

lopezvoliver commented on geeet/tseb.py:
```diff
@@ -218,10 +219,16 @@ def tseb_series(img=None,    # ee.Image with inputs as bands (takes precedence o
+            # Retrieve latitude and longitude from spatial dimensions
+            spatial_dims = find_spatial_dims(img)
+            if len(spatial_dims) == 2:
+                latitude = latitude if latitude is not None else img[spatial_dims[0]]
```
Sorry for the long wait! I only updated these two lines to give preference to the `longitude` and `latitude` keyword arguments if provided. If you run Notebook 05, which has `x` and `y` dimensions in a local projection, and you use these as lon/lat, the result is not as expected:

<img width="905" height="290" alt="nb5_bad_output" src="https://github.com/user-attachments/assets/a2d54a3a-0102-4e09-9ab7-5ba95cf88068" />

Here's the expected result which we get if we use the correct `longitude` and `latitude` values defined in the notebook:

<img width="905" height="290" alt="nb5_good_output" src="https://github.com/user-attachments/assets/ca29929b-5452-4380-8ece-c2630ad1c1f2" />

lopezvoliver commented:
Thanks @dobedobedo ! and sorry for the long delay

Cheers
